LambdaEngine/Include/Input/API/InputActionSystem.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include "Input/API/InputState.h"
+#include "Containers/THashTable.h"
+
+namespace LambdaEngine
+{
+	enum EAction {
+		PLAYER_FORWARD,
+		PLAYER_BACKWARD,
+		PLAYER_LEFT,
+		PLAYER_RIGHT,
+		// more actions to come...
+	};
+
+	class LAMBDA_API InputActionSystem
+	{
+	public:
+		DECL_STATIC_CLASS(InputActionSystem);
+
+		// TODO: Read keymapping from file
+		static bool IsActive(EAction action);
+
+		FORCEINLINE static InputActionSystem* GetDevice()
+		{
+			return s_pInputActionSystem;
+		}
+
+	private:
+		static InputActionSystem* s_pInputActionSystem;
+		static std::unordered_map<EAction, EKey> m_KeyMapping;
```

Comment:
* Why are we storing an instance of this class, since the only thing that it stores is m_KeyMapping, which in itself is already a static member? It seems like you have mixed up a singleton and static class. The difference is that a singleton is written like it is a standard object, but only one instance exists. A singleton is similar to a "normal" class, except for private constructors to prevent that anyone outside of the class creates an instance. The single object is then created somehow with a static function, either by using a pair of functions (Initialize for creating and Release for destroying) or use a single function that returns a static instance.